Components/Queues/Drivers/PhpAmqpLibDriver.php

```diff
@@ -0,0 +1,429 @@
+<?php
+
+
+namespace Smartbox\Integration\FrameworkBundle\Components\Queues\Drivers;
+
+use PhpAmqpLib\Channel\AMQPChannel;
+use PhpAmqpLib\Exchange\AMQPExchangeType;
+use PhpAmqpLib\Message\AMQPMessage;
+use PhpAmqpLib\Wire\AMQPTable;
+use Smartbox\CoreBundle\Type\SerializableInterface;
+use Smartbox\Integration\FrameworkBundle\Components\Queues\PhpAmqpLibQueueManager;
+use Smartbox\Integration\FrameworkBundle\Components\Queues\QueueMessage;
+use Smartbox\Integration\FrameworkBundle\Components\Queues\QueueMessageInterface;
+use Smartbox\Integration\FrameworkBundle\Core\Messages\Context;
+use Smartbox\Integration\FrameworkBundle\DependencyInjection\Traits\UsesSerializer;
+use Smartbox\Integration\FrameworkBundle\Exceptions\Handler\UsesExceptionHandlerTrait;
+use Smartbox\Integration\FrameworkBundle\Service;
+
+/**
+ * Class PhpAmqpLibDriver
+ * @package Smartbox\Integration\FrameworkBundle\Components\Queues\Drivers
+ */
+class PhpAmqpLibDriver extends Service implements PurgeableQueueDriverInterface
+{
+    use UsesSerializer;
+    use UsesExceptionHandlerTrait;
+
+    /**
+     * Maximum amount of time (in seconds) to wait for a message.
+     */
+    const WAIT_TIMEOUT = 10;
+
+    /**
+     * Maximum amount of time (in seconds) to set as an interval while consuming messages
+     */
+    const WAIT_PERIOD = 0.2;
+
+    /**
+     * Represents the amount of message to consume by iteration
+     */
+    const PREFETCH_COUNT = 1;
+
+    /*
+     * To use the default exchange pass an empty string
+     */
+    const EXCHANGE_NAME = '';
+
+
+    /**
+     * @var \AMQPQueue
+     */
+    private $queue;
+
+    /**
+     * @var int Time in ms
+     */
+    private $dequeueingTimeMs = 0;
+
+    /**
+     * @var string
+     */
+    private $format;
+
+    /**
+     * @var PhpAmqpLibQueueManager
+     */
+    private $manager;
+
+    /**
+     * @var AMQPChannel
+     */
+    private $channel;
+
+    /**
+     * PhpAmqpLibDriver constructor.
+     * @param QueueManager $manager
+     */
+    public function __construct(PhpAmqpLibQueueManager $manager, string $format = null)
+    {
+        parent::__construct();
+        $this->manager = $manager;
+        $this->format = $format ?? self::FORMAT_JSON;
+    }
+
+    /**
+     * PhpAmqpLibDriver destruct
+     */
+    final function __destruct()
+    {
+        try {
+            $this->manager->disconnect();
+        } catch (\Exception $connectionException) {
+            $this->getExceptionHandler()($connectionException);
```

Comment:
This piece of code is going to fail because the re is an extra parenthesis `getExceptionHandler()($connectionException)`
Another question is, what did it change in the manager disconnect method that we need to use the `getExceptionHandler`?

Reply:
It was modified.